internal/pkg/util/env/env.go

```diff
@@ -14,6 +14,16 @@ import (
 	"github.com/sylabs/singularity/v4/internal/pkg/util/shell/interpreter"
 )
 
+var readonlyVars = map[string]bool{
+	"GID":    true,
```

Comment:
Nit: the order here is a bit weird - it looks like the items were alphabetized and _then_ EUID was added? Perhaps EUID can be moved to the top so the list remains alphabetized (for readability/maintenance purposes; obviously this is of no significance in-memory, when it comes to a map).

Reply:
Yeah, that's exactly what I did I think. :-)